### DISCUSSION TITLE
SOFAR 6KTLM-G3-A / 23XXXXXXXX / LSW3_15_270A_1.38

### DISCUSSION THREAD

**Discussion by gushcs** [StephanJoubert/home_assistant_solarman#272] [Show and tell]
Despite not knowing much about coding, I managed to put together a setup for my SOFAR inverter. I hope it can help other people with the same problem.

https://gist.github.com/gushcs/223ca2c61af2d345238d286746617650

**Comment by doughBR** [comment]
Thanks! Works in my system.

INVERSOR
Model: 10.5KTLM-G3
SN: ST1ESxxxxxxxxx
Protocol Version:  1.07

DATALOGGER
Model: LSW3_15_270A_1.68
SN: 27xxxxxxxx

**Reply by stephencrowther** [reply]
Thank you so much for this have been trying and failing to get my 5KTLM G3 inverter working for ages with the base definitions files but none of them worked.

This one seems to work for me now! Thanks again!

**Comment by janik-design** [comment]
Helo.
I'm nob and I got a stupid question:
How can I add this yaml file?
Solarman doesn't work with my Sofar and I need to try this.

**Reply by davidrapan** [reply]
Hi @janik-design, if you still have any problems you can try my integration: https://github.com/davidrapan/ha-solarman

If you experience any issues with installation for your device you can reach me there through discussions or issues. 😉

**Reply by stephencrowther** [reply]
> Helo. I'm nob and I got a stupid question: How can I add this yaml file? Solarman doesn't work with my Sofar and I need to try this.

Easiest way is to copy all the text out of this file and copy it into the custom yaml file in the solarman custom config folder.

**Comment by ghedisejn** [comment]
Yes!! Thanks a lot! I didn't found the costum_parameters.yaml so i downloaded the sofar_hyd3k-6k-es.yaml file and copy the code in to it, afterwards upload it again --> that worked
